commit f6c3146229b4b3c40402db396d7d5002bba011df (refs/changes/48/3048/1)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2017-12-06T01:29:37+00:00 (1 year, 10 months ago)
    
    OS-6497 Non-primary NICs of "joyent" branded zones dno't work with just "addrconf"

diff --git a/src/vm/node_modules/VM.js b/src/vm/node_modules/VM.js
index 3f1e0bd4..e183056c 100644
--- a/src/vm/node_modules/VM.js
+++ b/src/vm/node_modules/VM.js
@@ -3247,6 +3247,10 @@ function writeZoneNetfiles(payload, log, callback)
                         static_ips);
                 }
 
+                if (addrconf_found) {
+                    replaceFile(zoneroot, '/etc/addrconf.' + n.interface, '');
+                }
+
                 if (n.hasOwnProperty('primary') && !primary_found) {
                     // Only allow one primary network
                     primary_found = true;
@@ -3257,10 +3261,6 @@ function writeZoneNetfiles(payload, log, callback)
                     if (dhcp_found) {
                         replaceFile(zoneroot, '/etc/dhcp.' + n.interface, '');
                     }
-                    if (addrconf_found) {
-                        replaceFile(zoneroot, '/etc/addrconf.' + n.interface,
-                            '');
-                    }
                 }
             });
         }
